File: src/main/java/org/cytoscape/biogwplugin/internal/query/QueryTemplate.java
package org.cytoscape.biogwplugin.internal.query;

import java.util.ArrayList;

public class QueryTemplate {
	String name;
	String description;
	String sparqlString;
	ArrayList<QueryParameter> parameters;
	public QueryTemplate(String name, String description, String sparqlString) {
		super();
		this.name = name;
		this.description = description;
		this.sparqlString = sparqlString;
		this.parameters = new ArrayList<>();
	}
	
	public void addParameter(QueryParameter p) {
		parameters.add(p);
	}
	
}

File: src/main/java/org/cytoscape/biogwplugin/internal/query/QueryTemplate.kt
package org.cytoscape.biogwplugin.internal.query

import org.cytoscape.biogwplugin.internal.parser.BGReturnType
import java.util.ArrayList

class QueryTemplate(var name: String, var description: String, var sparqlString: String, var returnType: BGReturnType) {

    var parameters = ArrayList<BGQueryParameter>()

    fun addParameter(p: BGQueryParameter) {
        parameters.add(p)
    }
}